pkg/agent/client.go

```diff
@@ -439,6 +449,9 @@ func (a *Client) Serve() {
 				dialResp.GetDialResponse().ConnectID = connID
 				if err := a.Send(dialResp); err != nil {
 					klog.ErrorS(err, "could not send dialResp")
+					// clean-up is normally called from remoteToProxy which we will never invoke.
+					// So we are invoking it here to force the clean-up to occur.
+					connCtx.cleanup()
```

Comment:
Need to close dialDone *before* calling cleanup, right now we're calling it after.